### ISSUE TITLE
Fingerprints for items

### ISSUE THREAD

**Issue by BernhardPosselt** [fguillot/picoFeed#273]
Let's say I want to filter out items that are the same. Prime example here are Newspaper feeds where there is usually a main feed and more feeds based on categories. The main feed will also contain items which appear in the category feeds. 

However as a user, this makes feed reading really tedious since you've got lots of duplicates. 

My current fix for this issue is to generate a hash over title, link, body and enclosures. When displaying a feed, I filter out duplicates. Also when marking an item read I'm doing this for all other items with the same hash.

However there are some problems with this: some feeds (slashdot.org, derstandard.at) ship the same content with different URL. However you can't exlude the URLs from the hash since this will break all feeds that contain items which only differ only by URL.

So basically I'd need a default implementation which you could override. Should we include this in picofeed?

@fguillot @mkresin

**Comment by BernhardPosselt** [comment]
Implementation would probably be similar to the rules, use a default implementation or load functions from files like **slashdot.org.php** which contain something like this:

```php
return function ($item, $feed) {
    return md5($item->getTitle() . $item->getContent());
}

**Comment by fguillot** [comment]
I'm curious to see how other RSS librairies handle this situation.

If we put that in rule files, someone have to maintain it.

**Comment by BernhardPosselt** [comment]
No worries, I'd maintain and implement the feature :) 

My experience from ditching my own scraper system in favor of picofeed's has been very positive and the main idea behind putting this into picofeed is to get more contributions from others.

I don't think any other RSS/Atom library provides this feature since it's not related to RSS/Atom parsing but rather feed readers. I don't think however that there's a better way than a hash to find duplicates. Backwards compatability is not harmed and the deduplication handling is entirely up to the implementor.

**Comment by cwmke** [comment]
Tiny Tiny RSS currently has a built-in plugin called [af_psql_trgm](https://git.framasoft.org/framasoft/framanews_ttrss/blob/117efb6fa497ce873c9e18368d2a9fc61eaa8ef8/plugins/af_psql_trgm/init.php) which is the only other reader I have used with this functionality. It's tied to postgresql though since it uses pg_trgm for finding similar articles. I found it very effective while using it.
